packages/gitmoji-changelog-cli/src/cli.js

```diff
@@ -36,7 +46,22 @@ async function main(options = {}) {
   }
 
   try {
-    const changelog = await generateChangelog(options)
+    const tags = await gitSemverTagsAsync()
+    const lastTag = head(tags)
+
+    const from = options.mode === 'init' ? '' : lastTag
+    const packageInfo = await getPackageInfo()
+
+    let to = options.release === 'from-package' ? packageInfo.to : options.release
+    if (to && to !== 'next') {
+      if (!semver.valid(to)) {
+        throw new Error(`${to} is not a valid semver to.`)
+      }
+
+      to = sanitizeVersion(to)
+    }
+
+    const changelog = await generateChangelog({ ...options, from, to })
```

Comment:
**How it's done, it handles 4 cases:**
- **mode init:**  from: first tag, to: last tag
- **mode update & release manual version:** from: last tag, to: manual version
- **mode update & release from-package:** from: last tag, to: package version
- **mode update & release next:** from: last tag, to: next

It will not work for the last 3 update modes because it will look for tags between the last tag and an unknown version:

```
Error: fatal: ambiguous argument 'v1.0.1..next': unknown revision or path not in the working tree.
```

Reply:
So I think, we have to use different concept in options:
- `to` an `from` should always be a real tag
    - `to` should be a empty if we start from beginning (mode `init`)
    - `from` should be empty if we target a new version (manual, from-package or next)
    - we should also catch the error if the tags are not found.
- `newVersion` should be:
    - empty if mode `init`
    - `x.x.x` if manual version
    - `x.x.x` if from-package
    - `next` if it's for next version

So we could call the function by doing:
```js
await generateChangelog(from, to, options)
```
`options` should have only necessary options for the core: 
```js
const options = { newVersion, groupSimilarCommits }
```

Reply:
I agree about rethinking the CLI parameters because this is not straight forward right now. And it seems that this is not "replayable".

But I think this is something to do in an other PR.
This one is about changing core API.

Reply:
Agree, but here I was talking about change core API parameters and options here, not the CLI:
```
await generateChangelog(from, to, options)
```

Reply:
Yeah, when I wrote lasts PRs, I was thinking to rename all function with this signature:
 - ` const generateChangelog = options => (from, to) => {}`

Because right know, we don't know **what** we destructure, sometimes this is options, sometimes not.
This not easy to read right know for me.